src/core/structure/sheet.ts

```diff
@@ -396,18 +435,114 @@ export default class Sheet {
     return this.cellData.get(cellKey)!;
   }
 
-  private resolveCell(cell: Cell): boolean {
-    const value = this.expressionHandler.evaluate(cell.formula);
-    if (value == null) {
-      cell.state = CellState.INVALID_EXPRESSION;
-      return false;
+  /**
+   * Returns true if the value of the cell has changed.
+   */
+  private resolveCell(cell: Cell, colKey: ColumnKey, rowKey: RowKey): boolean {
+    const evalResult = this.expressionHandler.evaluate(cell.formula);
+    if (!evalResult) {
+      const prevState = cell.state;
+      cell.setState(CellState.INVALID_EXPRESSION);
+      return prevState == CellState.OK; // Consider the cell's value changed if its state changes from OK to invalid.
     }
 
     // TODO Resolve restrictions from cell formatting here (CellState.INVALID_FORMAT).
 
-    cell.state = CellState.OK;
-    cell.value = value;
-    return cell.formula != cell.value;
+    cell.setState(CellState.OK);
+
+    const valueChanged = cell.value != evalResult.value;
+    cell.value = evalResult.value;
+    this.updateCellReferences(cell, colKey, rowKey, valueChanged, evalResult);
```

Comment:
This function internally resolve and emit functions, this name is under playing how many things this function is doing. I would suggest `UpdateAndEvaludateCellReferences`

Also, it is not ideal to have a function with that many parameters.

Reply:
> this name is under playing how many things this function is doing

I've renamed it to `handleCellReferenceChanges`; I think along with the comment this is descriptive enough.



> Also, it is not ideal to have a function with that many parameters.

Yes, but we need to pass a lot of context to the method. It's a result of splitting the method into two, which I think is still better.